Plug-Ins/Dialog Editor/DialogEditorWindowController.swift

```diff
@@ -0,0 +1,1156 @@
+		NotificationCenter.default.addObserver(self, selector: #selector(selectedItemDidChange(_:)), name: DITLDocumentView.selectionDidChangeNotification, object: self.scrollView.documentView)
+		NotificationCenter.default.addObserver(self, selector: #selector(selectedItemWillChange(_:)), name: DITLDocumentView.selectionWillChangeNotification, object: self.scrollView.documentView)
+		self.loadItems()
+		self.updateView()
+	}
+	
+	@objc func itemDoubleClicked(_ notification: Notification) {
+		print("double clicked.")
+	}
+	
+	@objc func itemFrameDidChange(_ notification: Notification) {
+		print("resized/moved.")
+		self.setDocumentEdited(true)
+	}
+	
+	func reflectSelectedItem() {
+		for item in items {
+			if item.itemView.selected {
+				typePopup.selectItem(withTag: Int(item.itemType.rawValue))
+				typePopup.isEnabled = true
+				enabledCheckbox.state = item.enabled ? .on : .off
+				enabledCheckbox.isEnabled = true
+
+				switch item.itemType {
+				case .userItem, .unknown:
+					tabView.selectTabViewItem(at: 3)
+				case .helpItem:
+					tabView.selectTabViewItem(at: 2)
+					helpResourceIDField.integerValue = item.resourceID
+					helpTypePopup.selectItem(withTag: Int(item.helpItemType.rawValue))
+					if item.helpItemType == .HMScanAppendhdlg {
+						helpItemField.integerValue = Int(item.itemNumber)
+						helpItemField.isEnabled = true
+					} else {
+						helpItemField.isEnabled = false
+					}
+				case .button, .checkBox, .radioButton, .staticText, .editText:
+					tabView.selectTabViewItem(at: 0)
+					titleContentsField.stringValue = item.itemView.title
+				case .control, .icon, .picture:
+					tabView.selectTabViewItem(at: 1)
+					resourceIDField.integerValue = item.resourceID
+				}
+				return
+			}
+		}
+		tabView.selectTabViewItem(at: 3)
+		enabledCheckbox.isEnabled = false
+		typePopup.isEnabled = false
+	}
+	
+	@objc func selectedItemWillChange(_ notification: Notification) {
+		window?.makeFirstResponder(scrollView.documentView)
+	}
+	
+	@objc func selectedItemDidChange(_ notification: Notification) {
+		reflectSelectedItem()
+	}
+	
+	/// Reload the views representing our ``items`` list.
+	private func updateView() {
+		for view in self.scrollView.documentView?.subviews ?? [] {
+			view.removeFromSuperview()
+		}
+		var maxSize = NSSize(width: 128, height: 64)
+		for item in items {
+			self.scrollView.documentView?.addSubview(item.itemView)
+			let itemBox = item.itemView.frame
+			maxSize.width = max(itemBox.maxX, maxSize.width)
+			maxSize.height = max(itemBox.maxY, maxSize.height)
+		}
+		var documentBox = self.scrollView.documentView?.frame ?? NSZeroRect
+		documentBox.size.width = max(documentBox.size.width, maxSize.width + 16)
+		documentBox.size.height = max(documentBox.size.height, maxSize.height + 16)
+		self.scrollView.documentView?.frame = documentBox
+		self.scrollView.documentView?.clipsToBounds = true
```

Comment:
Looks like `clipsToBounds` requires the macOS 14 SDK. It's not a property I'm familiar with, but is there another way to achieve what you need here?